### PR TITLE
Added support for libld.so.2

### PR DESCRIPTION
**Pull request by rwmcintosh**

### CODE REVIEW

File: DynamicInterop/UnixLibraryLoader.cs [hunk 1]
```diff
@@ -14,7 +14,15 @@ internal class UnixLibraryLoader : IDynamicLibraryLoader
...
         {
-            return InternalLoadLibrary(filename);
+            const int RTLD_LAZY = 0x1;
+
+            if (_so == 0)
+                return InternalLoadLibrary(filename, RTLD_LAZY);
+
+            if (_so == 1)
+                return dlopen1(filename, RTLD_LAZY);
+            else
```
Comment (Yuri05) [inline DynamicInterop/UnixLibraryLoader.cs:24]: `else `is redundant here

File: DynamicInterop/UnixLibraryLoader.cs [hunk 2]
```diff
@@ -23,7 +31,13 @@ public IntPtr LoadLibrary(string filename)
         /// <returns>The last error.</returns>
         public string GetLastError()
         {
-            return dlerror();
+            if (_so == 0)
+                throw new Exception("Called GetLastError before LoadLibrary");
+
+            if (_so == 1)
+                return dlerror1();
+            else
```
Comment (Yuri05) [inline DynamicInterop/UnixLibraryLoader.cs:39]: same

File: DynamicInterop/UnixLibraryLoader.cs [hunk 3]
```diff
@@ -33,34 +47,48 @@ public string GetLastError()
         /// <returns>True if the function dlclose returned 0</returns>
         public bool FreeLibrary(IntPtr handle)
         {
+            if (_so == 0)
+                throw new Exception("Called GetLastError before LoadLibrary");
```
Comment (Yuri05) [inline DynamicInterop/UnixLibraryLoader.cs:51]: "Called ~GetLastError~ FreeLibrary before LoadLibrary"

File: DynamicInterop/UnixLibraryLoader.cs [hunk 4]
```diff
@@ -33,34 +47,48 @@ public string GetLastError()
...
+                var status = dlclose2(handle);
+                return status == 0;
+            }
         }
 
         public IntPtr GetFunctionAddress(IntPtr hModule, string lpProcName)
         {
-            return dlsym(hModule, lpProcName);
+            if (_so == 0)
+                throw new Exception("Called GetLastError before LoadLibrary");
```
Comment (Yuri05) [inline DynamicInterop/UnixLibraryLoader.cs:70]: "Called ~GetLastError~ GetFunctionAddress before LoadLibrary"

**Comment by rwmcintosh** [comment]
Allowed VS to update some project settings as well once I opened the project.

**Review by Yuri05 — APPROVED** [review]

**Pull request merged**